From cbe4ab7294afc4afca0d8d1f9562e973ca8a1810 Mon Sep 17 00:00:00 2001
From: Dan Winship <danw@gnome.org>
Date: Sat, 06 Aug 2011 00:04:03 +0000
Subject: gnutls: override minimum key length

By default, gnutls requires server keys to be a reasonable size and
will return an error rather than connecting to a server with an
insecurely small key. But other TLS libraries don't do this, and some
web servers (especially embedded ones) use small keys. So bump down
the limit.

https://bugzilla.gnome.org/show_bug.cgi?id=652284
---
diff --git a/tls/gnutls/gtlsconnection-gnutls.c b/tls/gnutls/gtlsconnection-gnutls.c
index 48e05d6..da3f76e 100644
--- a/tls/gnutls/gtlsconnection-gnutls.c
+++ b/tls/gnutls/gtlsconnection-gnutls.c
@@ -253,6 +253,11 @@ g_tls_connection_gnutls_initable_init (GInitable     *initable,
       return FALSE;
     }
 
+  /* Some servers (especially on embedded devices) use tiny keys that
+   * gnutls will reject by default. We want it to accept them.
+   */
+  gnutls_dh_set_prime_bits (gnutls->priv->session, 256);
+
   gnutls_transport_set_push_function (gnutls->priv->session,
 				      g_tls_connection_gnutls_push_func);
   gnutls_transport_set_pull_function (gnutls->priv->session,
--
cgit v0.9.0.2
